### DISCUSSION TITLE
Distribution of Transaction Latencies

### DISCUSSION THREAD

**Discussion by ckolovson** [TPC-Council/HammerDB#269] [Q&A] [answered]
I am just starting to get familiar w/ HammerDB and am comparing it with sysbench for use with MariaDB and its SaaS product (SkySQL). I notice that HammerDB does not appear to have a way of measuring and displaying latencies and/or showing a histogram of latencies as sysbench supports. I found this old discussion thread on sourceforge from 2016: 
https://sourceforge.net/p/hammerdb/discussion/292312/thread/62a92fb0/
Is it still the case that HammerDB has no facility for measuring and displaying query latencies and/or latency distributions?

**Comment by sm-shaw** [comment, marked as answer]
HammerDB calls this time profiling. There are 2 packages to do time profiling. By default, from 4.1 "xtprof" is used to profile the response times from all virtual users as described here https://www.hammerdb.com/blog/uncategorized/hammerdb-v4-1-extended-time-profiling/.
Also, there is an alternative package called "etprof" to profile the first virtual user only but to sample at time intervals as described here. 
https://www.hammerdb.com/blog/uncategorized/how-to-graph-hammerdb-response-times/
How to use them and switch which one to use is described in the documentation here: https://www.hammerdb.com/docs/ch04s06.html

**Reply by ckolovson** [reply]
Thanks!!
